sip-ri/src/test/java/gov/nist/javax/sip/parser/extensions/DiversionParserTest.java

```diff
@@ -0,0 +1,40 @@
+package gov.nist.javax.sip.parser.extensions;
+
+import gov.nist.javax.sip.header.extensions.DiversionHeader;
+import gov.nist.javax.sip.header.extensions.DiversionList;
+import gov.nist.javax.sip.parser.ParserTestCase;
+
+
+import java.text.ParseException;
+
+public class DiversionParserTest extends ParserTestCase {
+
+    public void testParser() {
+        String[] diversions = {
+            "Diversion: sip:user@example.com;reason=busy;limit=4;privacy=conditionally;counter=2;screen=yes;extension=token\n"
+        };
+        super.testParser(DiversionParser.class, diversions);
+    }
+
+    public void testParseDiversionHeaderParameters() throws ParseException {
+        String diversionHeader = "Diversion: sip:user@example.com;reason=busy;limit=4;privacy=conditionally;counter=2;screen=yes;extension=token\n";
+        try {
+            DiversionParser parser = new DiversionParser(diversionHeader);
+            DiversionList diversionList = (DiversionList) parser.parse();
+            assertNotNull(diversionList);
+
+            for (DiversionHeader diversion : diversionList) {
```

Comment:
You should not only print them but validate them with assert to ensure they match the value that was passed in the initial diversionHeader String and test the toString()/encode() too to ensure it properly output the same String